pkg/endpoint/bpf.go

```diff
@@ -692,18 +695,24 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 		c.Mutex.RUnlock()
 	}
 
+	// Generate header file specific to this endpoint for use in compiling
+	// BPF programs for this endpoint.
 	if err = e.writeHeaderfile(epdir, owner); err != nil {
 		e.Mutex.Unlock()
 		return 0, fmt.Errorf("Unable to write header file: %s", err)
 	}
 
+	// Cache endpoint information
+	// TODO (ianvernon): why do we need to do this?
```

Comment:
In principle I think it's caching endpoint information before performing calculation in case things change, but when I actually look at it, it appears to be a representation of the endpoint's IP, MAC, etc. which I would not expect to change mid-policy computation (or, for that matter, at all). @jrajahalme ?

Reply:
Yeah, it wasn't self-evident to me as to why we were doing this. I'm hoping that documentation of what the code is doing in-line will reveal whether we can optimize it further.